### PR TITLE
Move ValidateWebhook from Hook to Github

### PR DESCRIPTION
**Pull request by ashleydavies**
Resolves a TODO in `needs-rebase` by removing a dependency from external-plugins to Hook. This will cause some conflict with another PR I have open but I'll rebase that onto this if these changes are wanted.

We're hoping to upstream a Webhook -> GCS/AWS bucket dumper which we discussed a few months ago in Slack very shortly (https://kubernetes.slack.com/archives/C09QZ4DQB/p1524580510000898), which this is a light prerequisite for (to avoid another component with a dependency on Hook), and I'd rather merge this separately to keep that PR focused.

cc @petemounce @DoomGerbil @efarem @lottec @majolo

### CODE REVIEW

File: prow/github/webhooks.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
+package github
+
+import (
+	"github.com/sirupsen/logrus"
```
Comment (kargakis) [inline prow/github/webhooks.go]: nit: separate imports into logical groups (stdlib, third-party, test-infra)

**Comment by kargakis** [comment]
/approve

seems you have an issue with missing boilerplate

**Comment by kargakis** [comment]
lgtm, please squash to a single commit and I will tag it

**Comment by ashleydavies** [comment]
Thanks @kargakis! I'll sort out the other PR tomorrow.

**Comment by cjwagner** [comment]
This change broke `needs-rebase` because it uses a plugin agent to load the plugin configuration which fails when it finds unknown plugin names. The plugin names are unknown because they are no longer compiled into the binary.
This change will need to be accompanied by some change to the plugin agent (or needs-rebase) to make it possible to load the plugin configuration without having all plugins compiled in.

I'm going to revert this for now so that the we can continue with the roll out of the improved job config defaulting.

**Comment by ashleydavies** [comment]
Apologies @cjwagner! I'll update this change shortly to keep the import or remove the dependecy

**Pull request merged**